### PR TITLE
Feat/doc update

### PR DESCRIPTION
**Pull request by csumudu**

### CODE REVIEW

**Review by csumudu — COMMENTED** [review]
```diff
+import {
+  DataItem,
+  SortChange,
+  SortChangeType,
+} from "../../../../shared/models";
+
+export function* heapSort(
+  col: Array<DataItem<number>>
+): Generator<SortChange> {
+  const source = [...col];
+  const n = source.length;
+
+  // Build max heap
+  for (let i = Math.floor(n / 2) - 1; i >= 0; i--) {
+    yield* heapify(source, n, i);
+  }
+
+  // Extract elements from heap one by one
+  for (let i = n - 1; i > 0; i--) {
+    // Move current root to end
+    const temp = source[0];
+    source[0] = source[i];
+    source[i] = temp;
+
+    yield {
+      type: SortChangeType.SWAP,
+      changes: {
+        [0]: source[0],
+        [i]: source[i],
+      },
+    };
+
+    // Mark the element as sorted
+    yield {
+      type: SortChangeType.SORTED,
+      changes: {
+        sortedIndex: i,
+      },
+    };
+
+    // Call heapify on the reduced heap
+    yield* heapify(source, i, 0);
+  }
+
+  // Mark the first element as sorted (it's the smallest)
+  yield {
+    type: SortChangeType.SORTED,
+    changes: {
+      sortedIndex: 0,
+    },
+  };
+}
+
+function* heapify(
+  arr: Array<DataItem<number>>,
+  n: number,
+  i: number
+): Generator<SortChange> {
+  let largest = i; // Initialize largest as root
+  const left = 2 * i + 1; // left child
+  const right = 2 * i + 2; // right child
+
+  // Highlight the current node being processed
+  yield {
+    type: SortChangeType.PROCESSING,
+    changes: {
+      [i]: arr[i],
+    },
+  };
+
+  // If left child is larger than root
+  if (left < n && arr[left].value > arr[largest].value) {
+    largest = left;
+  }
+
+  // If right child is larger than largest so far
+  if (right < n && arr[right].value > arr[largest].value) {
+    largest = right;
+  }
+
+  // If largest is not root
+  if (largest !== i) {
+    const temp = arr[i];
+    arr[i] = arr[largest];
+    arr[largest] = temp;
+
+    yield {
+      type: SortChangeType.SWAP,
+      changes: {
+        [i]: arr[i],
+        [largest]: arr[largest],
+      },
+    };
+
+    // Recursively heapify the affected sub-tree
+    yield* heapify(arr, n, largest);
+  }
+}
```
- Good use of generators to represent state changes for visualization; this matches the existing pattern in the project.
- `heapify` implementation correctly yields a PROCESSING event before comparisons; this supports visual highlighting.
- Swap and recursive calls in `heapify` maintain heap property effectively.
- After each root swap and sorting of the element, yielding a SORTED event is consistent with visual marking of sorted elements.
- Consider adding comments to clarify that the last yield in `heapSort` marks index 0 as sorted since it remains.
- The `source` local copy is modified; ensure no side effects on the original `col` (seems intentional and good).
- Overall, the heap sort algorithm implementation is correct and well adapted for visualization needs.

```diff
+import {
+  SortChange,
+  DataItem,
+  SortChangeType,
+} from "../../../../shared/models";
+
+export const heapSortVisualizer = (
+  value: SortChange,
+  update: (id: string, val: Partial<DataItem<number>>) => void,
+  data: DataItem<number>[]
+) => {
+  if (value.type === SortChangeType.SWAP) {
+    const changes = value.changes;
+
+    Object.keys(changes).forEach((key) => {
+      const item = data[+key];
+      const newValue = changes[+key];
+
+      update(item.id, {
+        value: newValue.value,
+        borderColor: "border-orange-500",
+      });
+    });
+  }
+
+  if (value.type === SortChangeType.PROCESSING) {
+    const changes = value.changes;
+
+    Object.keys(changes).forEach((key) => {
+      const item = data[+key];
+
+      update(item.id, {
+        borderColor: "border-blue-500",
+        bgColor: "bg-blue-100",
+      });
+    });
+  }
+
+  if (value.type === SortChangeType.SORTED) {
+    const changes = value.changes;
+    const item = data[changes.sortedIndex];
+
+    update(item.id, {
+      isSorted: true,
+      borderColor: "border-green-500",
+      bgColor: "bg-green-100",
+    });
+  }
+};
```
- Visualization updates correctly handle the three key SortChangeTypes: SWAP, PROCESSING, and SORTED.
- Using distinct color classes (`border-orange-500`, `border-blue-500`, `border-green-500`) aligns with visual feedback from other sort visualizers.
- The code updates only the affected items as per the changes from the generator.
- Slight optimization: could consider resetting colors for non-affected items or managing "clear" states if necessary (might be handled elsewhere).
- Usage of `+key` to coerce string keys from Object.keys to number indices is appropriate for array access.
- Good to maintain consistent naming and structure with other visualizers.

```diff
-import { DataItem, SortAlgoTypes, SortChange } from "../../../shared/models";
-import { bubbleSort } from "../sort/bubble-sort/algo";
-import { bubbleSortVisualizer } from "../sort/bubble-sort/visualizer";
+import { DataItem, SortAlgoTypes, SortChange } from "../../../shared/models";
+import { bubbleSort } from "../sort/bubble-sort/algo";
+import { bubbleSortVisualizer } from "../sort/bubble-sort/visualizer";
+import { heapSort } from "../sort/heap-sort/algo";
+import { heapSortVisualizer } from "../sort/heap-sort/visualizer";
 import { insertionSort } from "../sort/insertion-sort/algo";
 import { insertionSortVisualizer } from "../sort/insertion-sort/visualizer";
 import { mergeSort } from "../sort/merge-sort/algo";
@@ -17,6 +19,8 @@ export class AlgorithmFactory {
     switch (type) {
       case SortAlgoTypes.BUBBLE_SORT:
         return bubbleSort;
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSort;
       case SortAlgoTypes.INSERTION_SORT:
         return insertionSort;
       case SortAlgoTypes.SELECTION_SORT:
@@ -36,6 +40,8 @@ export class AlgorithmFactory {
         return insertionSortVisualizer;
       case SortAlgoTypes.BUBBLE_SORT:
         return bubbleSortVisualizer;
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSortVisualizer;
       case SortAlgoTypes.SELECTION_SORT:
         return selectionSortVisualizer;
       case SortAlgoTypes.MERGE_SORT:
```
- Well integrated new heap sort algorithm and visualizer into the factory methods.
- The order consistent with other algorithms.
- Makes the heap sort visualizer and algo selectable via the existing sorting algorithm infrastructure.
- No obvious edge cases missed here.
- Confirmed imports paths are consistent; no typos.
- Could consider adding test coverage if not existing for the new algorithm.

```diff
-    <title>Algorithm Playground - CallStackSurfer </title>
+    <title>Algorithm Playground</title>
```
- Title simplified; removing '- CallStackSurfer' may reduce branding visibility but aligns with more concise titles.
- Also trimming trailing space in original.
- Ensure consistency with other metadata or documents if 'CallStackSurfer' branding is important.

```diff
+      <SortWidget title="Heap Sort" type={SortAlgoTypes.HEAP_SORT} />
```
- Added Heap Sort widget to the sorting dashboard, thus exposing the new algorithm in the UI.
- Ordering of sort widgets seems logical, placing Heap Sort near Merge and Quick sorts.
- Confirm that styling/layout accommodates the extra widget gracefully - grid definition suggests it will.
- Makes the new algorithm easily discoverable by users.

```diff
+  HEAP_SORT = "HEAP_SORT",
```
- Adds Heap Sort enum entry matching the naming convention.
- Good to keep this consistent; no trailing commas (styling consistent with project).
- Allows usage of this enum wherever sort algorithm types are referenced.
- Important to ensure this is fully synced with code factory, UI, and configuration (which seems done).
 
```diff
-# Algorithm Playground
-This application provides a visual representation of various algorithms, allowing users to observe their behavior in real time.
+# 🎯 Algorithm Playground
-
-It leverages Recoil for state management, ensuring efficient rendering by updating only the necessary components instead of re-rendering the entire dataset when visual changes are applied to data cells.
+A modern, interactive web application that provides **real-time visual representations** of various algorithms and data structures. Built with React, TypeScript, and Ant Design, this playground makes learning algorithms engaging and intuitive.
-
-Welcome pull requests to enhance the playground! You can contribute by adding new visualizations, as well as extending AlgorithmFactory class with new generator and visualizer functions.
\ No newline at end of file
+[![Live Demo](https://img.shields.io/badge/Live%20Demo-algo.callstacksurfer.com-blue?style=for-the-badge)](https://algo.callstacksurfer.com)
+
+## ✨ Features
+
+- **🎬 Real-time Visualization**: Watch algorithms execute step-by-step with smooth animations
+- **⚡ Optimized Performance**: Uses Recoil for efficient state management, updating only necessary components
+- **🎛️ Interactive Controls**: Play, pause, stop, and reload animations with intuitive controls
+- **📱 Responsive Design**: Works seamlessly across desktop and mobile devices
+- **🎨 Modern UI**: Clean, professional interface built with Ant Design and Tailwind CSS
+
+## 🧮 Supported Algorithms
+
+### Sorting Algorithms
+- **Bubble Sort** - Simple comparison-based sorting
+- **Selection Sort** - In-place comparison sorting algorithm
+- **Insertion Sort** - Builds sorted array one element at a time
+- **Merge Sort** - Efficient divide-and-conquer sorting
+- **Quick Sort** - Fast divide-and-conquer with pivot selection
+
+### Data Structures
+- **Stack** - LIFO (Last In, First Out) operations
+- **Queue** - FIFO (First In, First Out) operations
+- **Search** - Various search algorithm implementations
+
+## 🚀 Getting Started
+
+### Prerequisites
+- Node.js (v16 or higher)
+- npm or yarn
+
+### Installation
+
+1. **Clone the repository**
+   ''bash
+   git clone https://github.com/your-username/algorithm-playground.git
+   cd algorithm-playground
+   ''
+
+2. **Install dependencies**
+   ''bash
+   npm install
+   ''
+
+3. **Start the development server**
+   ''bash
+   npm run dev
+   ''
+
+4. **Open your browser**
+   Navigate to `http://localhost:5173` to view the application
+
+### Build for Production
+
+''bash
+npm run build
+npm run preview
+''
+
+## 🛠️ Tech Stack
+
+- **Frontend Framework**: React 18 with TypeScript
+- **State Management**: Recoil
+- **UI Components**: Ant Design
+- **Styling**: Tailwind CSS
+- **Routing**: React Router v7
+- **Build Tool**: Vite
+- **Code Quality**: ESLint with TypeScript support
+
+## 📁 Project Structure
+
+''
+src/
+├── features/
+│   ├── algorithms/          # Algorithm implementations
+│   │   ├── sort/           # Sorting algorithms
+│   │   ├── search/         # Search algorithms
+│   │   ├── stack/          # Stack data structure
+│   │   ├── queue/          # Queue data structure
+│   │   └── services/       # Algorithm factory and utilities
+│   └── app/                # App-level components
+├── shared/                 # Shared utilities and components
+│   ├── components/         # Reusable UI components
+│   ├── hooks/             # Custom React hooks
+│   └── utils/             # Utility functions
+├── state/                 # State management
+│   ├── atoms/             # Recoil atoms
+│   ├── selectors/         # Recoil selectors
+│   └── reducers/          # State reducers
+└── config/                # Application configuration
+''
+## 🤝 Contributing
+
+We welcome contributions! Here are several ways you can help improve the Algorithm Playground:
+
+### Adding New Algorithms
+
+1. **Create algorithm implementation**
+   ''typescript
+   // src/features/algorithms/[category]/[algorithm-name]/algo.ts
+   export class YourAlgorithm {
+     // Implementation here
+   }
+   ''
+
+2. **Create visualizer**
+   ''typescript
+   // src/features/algorithms/[category]/[algorithm-name]/visualizer.ts
+   export class YourAlgorithmVisualizer {
+     // Visualization logic here
+   }
+   ''
+
+3. **Extend AlgorithmFactory**
+   ''typescript
+   // Add your algorithm to the factory
+   ''
+
+### Development Guidelines
+
+- Follow TypeScript best practices
+- Use existing component patterns
+- Ensure responsive design
+- Add proper error handling
+- Write clear, documented code
+
+### Submitting Changes
+
+1. Fork the repository
+2. Create a feature branch (`git checkout -b feature/amazing-algorithm`)
+3. Commit your changes (`git commit -m 'Add amazing new algorithm'`)
+4. Push to the branch (`git push origin feature/amazing-algorithm`)
+5. Open a Pull Request
+
+## 📝 Configuration
+
+The application can be configured through `src/config/app-config.ts`:
+
+''typescript
+export const APP_CONFIG = {
+  defaultCollectionSize: 100,    // Default array size
+  defaultIntervaleTimeout: 100,  // Animation speed (ms)
+  maxTimeout: 2000,             // Maximum delay
+  minTimeout: 0                 // Minimum delay
+}
+''
+
+## 🎮 Usage
+
+1. **Select an Algorithm**: Choose from sorting algorithms or data structures
+2. **Configure Parameters**: Adjust array size and animation speed
+3. **Start Visualization**: Click play to watch the algorithm in action
+4. **Control Playback**: Use pause, stop, or reload controls as needed
+
+## 📄 License
+
+This project is open source and available under the [MIT License](LICENSE).
+
+## 🌟 Acknowledgments
+
+- Built with ❤️ by [callstacksurfer.com](https://callstacksurfer.com)
+
+---
+
+**[🚀 Try it live at algo.callstacksurfer.com](https://algo.callstacksurfer.com)**
```
- README greatly improved with detailed sections: Features, Supported Algorithms, Getting Started, Tech Stack, and Contribution guidelines.
- Clear instructions with code blocks aid new contributors and users.
- Including the live demo badge and link is very helpful.
- The project structure is well documented, facilitating navigation.
- Nice to see explicit contribution flow for adding new algorithms.
- Minor note: typo in configuration key - `defaultIntervaleTimeout` probably should be `defaultIntervalTimeout` (typo "Intervale").
- Use of triple backticks (```) rather than double quotes (`''`) in code blocks would be more standard in markdown.
- The license and acknowledgment sections help with community and legal info.
- Consistent formatting with emoji icons adds user-friendly style.
- Overall, this will significantly enhance project onboarding and user experience.

```

**Review by csumudu — COMMENTED** [review]
```diff
@@ -1,169 +1,169 @@
-# 🎯 Algorithm Playground
-
-...
-
-## 🔥 Usage
-
-1. **Select an Algorithm**: Choose from sorting algorithms or data structures
-2. **Configure Parameters**: Adjust array size and animation speed
-3. **Start Visualization**: Click play to watch the algorithm in action
-4. **Control Playback**: Use pause, stop, or reload controls as needed
-
-## 📄 License
-
-This project is open source and available under the [MIT License](LICENSE).
-
-## 🌟 Acknowledgments
-
-- Built with ❤️ by [callstacksurfer.com](https://callstacksurfer.com)
-
--- 
-
-**[🚀 Try it live at algo.callstacksurfer.com](https://algo.callstacksurfer.com)**
+ # 🎯 Algorithm Playground
+
+ This README update is very comprehensive and well-structured.
+ - The introduction is clear and engaging; it highlights key technologies and features.
+ - Including badges and a live demo link at the top is a good UX practice.
+ - The feature list is detailed and uses appropriate emojis to improve readability.
+ - Supported algorithms section nicely categorizes sorting algorithms and data structures.
+ - Installation and getting started steps are straightforward and clear.
+ - Listing the tech stack and project structure adds helpful context for contributors.
+ - Contributing section is thorough and guides how to add algorithms and contribute.
+ - Configuration section clearly explains where to adjust app parameters.
+ - Usage instructions are intuitive.
+ - Licensing and acknowledgment sections are professional.
+
+Minor suggestions to consider:
+ - The block quoting for code snippets uses `''bash` instead of triple backticks with the language specifier. It is better to use triple backticks for consistency and GitHub flavored markdown compatibility.
+ - Some code snippets in installation commands are missing the closing triple backticks (for example, in the clone repository and install commands).
+ - The line breaks and indentation in `src/config/app-config.ts`'s code block seem slightly inconsistent with standard formatting; ensure the triple backticks are correctly placed.
+ - In "Supported Algorithms," consider listing HEAP_SORT since it was added.
+ - No visible typos or grammar issues.
+ - The new content significantly improves the project's documentation quality.
```


```diff
@@ -4,7 +4,7 @@
-    <title>Algorithm Playground - CallStackSurfer </title>
+    <title>Algorithm Playground</title> 
+// Removing " - CallStackSurfer" is okay if the project branding is more concise; consider SEO impact and whether you want consistent naming across pages.
```


```diff
@@ -1,6 +1,8 @@
-import { DataItem, SortAlgoTypes, SortChange } from "../../../shared/models";
-import { bubbleSort } from "../sort/bubble-sort/algo";
-import { bubbleSortVisualizer } from "../sort/bubble-sort/visualizer";
+import { DataItem, SortAlgoTypes, SortChange } from "../../../shared/models";
+import { bubbleSort } from "../sort/bubble-sort/algo";
+import { bubbleSortVisualizer } from "../sort/bubble-sort/visualizer";
+import { heapSort } from "../sort/heap-sort/algo";
+import { heapSortVisualizer } from "../sort/heap-sort/visualizer";
-import { insertionSort } from "../sort/insertion-sort/algo";
-import { insertionSortVisualizer } from "../sort/insertion-sort/visualizer";
-import { mergeSort } from "../sort/merge-sort/algo";
@@ -17,6 +19,8 @@ export class AlgorithmFactory {
-    switch (type) {
-      case SortAlgoTypes.BUBBLE_SORT:
-        return bubbleSort;
+    switch (type) {
+      case SortAlgoTypes.BUBBLE_SORT:
+        return bubbleSort;
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSort;
       case SortAlgoTypes.INSERTION_SORT:
         return insertionSort;
       case SortAlgoTypes.SELECTION_SORT:
@@ -36,6 +40,8 @@ export class AlgorithmFactory {
-        return insertionSortVisualizer;
-      case SortAlgoTypes.BUBBLE_SORT:
-        return bubbleSortVisualizer;
+        return insertionSortVisualizer;
+      case SortAlgoTypes.BUBBLE_SORT:
+        return bubbleSortVisualizer;
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSortVisualizer;
       case SortAlgoTypes.SELECTION_SORT:
         return selectionSortVisualizer;
       case SortAlgoTypes.MERGE_SORT:
```
- Good integration of heap sort and its visualizer into the factory.
- The imports are correct and maintain consistency.
- The switch cases appropriately include the new HEAP_SORT option.
- Suggestion: Consider adding comments or grouping cases for better maintainability if more algorithms are frequently added.
- Confirm the types and visualizer signatures match the existing factory interface.


```diff
@@ -0,0 +1,98 @@
+import {
+  DataItem,
+  SortChange,
+  SortChangeType,
+} from "../../../../shared/models";
+
+export function* heapSort(
+  col: Array<DataItem<number>>
+): Generator<SortChange> {
+  const source = [...col];
+  const n = source.length;
+
+  // Build max heap
+  for (let i = Math.floor(n / 2) - 1; i >= 0; i--) {
+    yield* heapify(source, n, i);
+  }
+
+  // Extract elements from heap one by one
+  for (let i = n - 1; i > 0; i--) {
+    // Move current root to end
+    const temp = source[0];
+    source[0] = source[i];
+    source[i] = temp;
+
+    yield {
+      type: SortChangeType.SWAP,
+      changes: {
+        [0]: source[0],
+        [i]: source[i],
+      },
+    };
+
+    // Mark the element as sorted
+    yield {
+      type: SortChangeType.SORTED,
+      changes: {
+        sortedIndex: i,
+      },
+    };
+
+    // Call heapify on the reduced heap
+    yield* heapify(source, i, 0);
+  }
+
+  // Mark the first element as sorted (it's the smallest)
+  yield {
+    type: SortChangeType.SORTED,
+    changes: {
+      sortedIndex: 0,
+    },
+  };
+}
+
+function* heapify(
+  arr: Array<DataItem<number>>,
+  n: number,
+  i: number
+): Generator<SortChange> {
+  let largest = i; // Initialize largest as root
+  const left = 2 * i + 1; // left child
+  const right = 2 * i + 2; // right child
+
+  // Highlight the current node being processed
+  yield {
+    type: SortChangeType.PROCESSING,
+    changes: {
+      [i]: arr[i],
+    },
+  };
+
+  // If left child is larger than root
+  if (left < n && arr[left].value > arr[largest].value) {
+    largest = left;
+  }
+
+  // If right child is larger than largest so far
+  if (right < n && arr[right].value > arr[largest].value) {
+    largest = right;
+  }
+
+  // If largest is not root
+  if (largest !== i) {
+    const temp = arr[i];
+    arr[i] = arr[largest];
+    arr[largest] = temp;
+
+    yield {
+      type: SortChangeType.SWAP,
+      changes: {
+        [i]: arr[i],
+        [largest]: arr[largest],
+      },
+    };
+
+    // Recursively heapify the affected sub-tree
+    yield* heapify(arr, n, largest);
+  }
+}
```
- Implementation follows a canonical heap sort algorithm correctly.
- The use of generators (`yield*`) to emit state changes aligns with the architecture.
- Good to emit `PROCESSING` SortChange before comparisons; this enables visualization of node highlighting.
- The handling of `SWAP` and `SORTED` states is consistent with other sorting algorithms.
- Marking the sorted element after each extraction is good for visualization.
- Minor note: The comment "Mark the first element as sorted (it's the smallest)" is more accurate as this is the last remaining element in the heap, which should indeed be sorted.
- Consider adding clearer comments on yielding `PROCESSING` to explain its role in visualizer semantics.
- Ensure that the DataItem array cloning (`const source = [...col];`) is deep enough depending on mutation needs — currently shallow cloning objects, but probably fine here based on architecture.

```diff
@@ -0,0 +1,49 @@
+import {
+  SortChange,
+  DataItem,
+  SortChangeType,
+} from "../../../../shared/models";
+
+export const heapSortVisualizer = (
+  value: SortChange,
+  update: (id: string, val: Partial<DataItem<number>>) => void,
+  data: DataItem<number>[]
+) => {
+  if (value.type === SortChangeType.SWAP) {
+    const changes = value.changes;
+
+    Object.keys(changes).forEach((key) => {
+      const item = data[+key];
+      const newValue = changes[+key];
+
+      update(item.id, {
+        value: newValue.value,
+        borderColor: "border-orange-500",
+      });
+    });
+  }
+
+  if (value.type === SortChangeType.PROCESSING) {
+    const changes = value.changes;
+
+    Object.keys(changes).forEach((key) => {
+      const item = data[+key];
+
+      update(item.id, {
+        borderColor: "border-blue-500",
+        bgColor: "bg-blue-100",
+      });
+    });
+  }
+
+  if (value.type === SortChangeType.SORTED) {
+    const changes = value.changes;
+    const item = data[changes.sortedIndex];
+
+    update(item.id, {
+      isSorted: true,
+      borderColor: "border-green-500",
+      bgColor: "bg-green-100",
+    });
+  }
+};
```
- Visualizer handles all major change types emitted by heapSort generator correctly.
- Using Tailwind CSS border and background color classes for visual indication is consistent and clear.
- Good that `update` is called with partial updates targeting just changed properties, enabling performant UI updates.
- Suggestion: It might help to reset colors of elements not being processed when changing states to avoid visual clashes or lingering highlights.
- Defensive programming could be used in case keys or indices are undefined or out of range, but likely the app ensures correct data.
- `bgColor` and `borderColor` updates help keep consistent UX.

```diff
@@ -6,6 +6,7 @@ const SortDashboardMain = () => {
-    <div className="grid gap-5 grid-cols-[repeat(auto-fit,minmax(30rem,1fr))] ">
-      <SortWidget title="Quick Sort" type={SortAlgoTypes.QUICK_SORT} />
-      <SortWidget title="Merge Sort" type={SortAlgoTypes.MERGE_SORT} />
+    <div className="grid gap-5 grid-cols-[repeat(auto-fit,minmax(30rem,1fr))] ">
+      <SortWidget title="Quick Sort" type={SortAlgoTypes.QUICK_SORT} />
+      <SortWidget title="Merge Sort" type={SortAlgoTypes.MERGE_SORT} />
+      <SortWidget title="Heap Sort" type={SortAlgoTypes.HEAP_SORT} />
       <SortWidget title="Bubble Sort" type={SortAlgoTypes.BUBBLE_SORT} />
       <SortWidget title="Insertion Sort" type={SortAlgoTypes.INSERTION_SORT} />
       <SortWidget title="Selection Sort" type={SortAlgoTypes.SELECTION_SORT} />
```
- Integration of Heap Sort widget into the sort dashboard is seamless.
- The title and type binding is appropriate.
- This enables Heap Sort to appear along with other sorting algorithms in the UI.
- Confirm that `SortWidget` properly supports the new type from enums.
- You might want to verify if sorting algorithm order here is intentional or if Heap Sort should be grouped or positioned differently.

```diff
@@ -64,6 +64,7 @@ export enum SortAlgoTypes {
-  INSERTION_SORT = "INSERTION_SORT",
-  QUICK_SORT = "QUICK_SORT",
-  MERGE_SORT = "MERGE_SORT",
+  INSERTION_SORT = "INSERTION_SORT",
+  QUICK_SORT = "QUICK_SORT",
+  MERGE_SORT = "MERGE_SORT",
+  HEAP_SORT = "HEAP_SORT",
 }
```
- Adding `HEAP_SORT` enum value is necessary and consistent with usage across factory, widgets, and visualizer.
- The new enum string is self-describing.
- It's good the enum stays alphabetically or logically sortable — you may decide if there's a preferred order (this looks fine).
```

**Review by csumudu — COMMENTED** [review]
```diff
+# 🎯 Algorithm Playground
+
+// Excellent detailed README update with clear sections and consistent formatting.
+// Adding badges, features, algorithms supported, setup guide, project structure, and contributing info improves usability significantly.
+// Suggest adding headers for the License file to ensure link visibility or ensuring it exists.
+// Check if all commands are using three backticks ```bash instead of two '', likely a typo in code fences (''bash vs ```bash).
+// In the Installation steps and code blocks, replace ''bash with ```bash for markdown consistency and proper code block rendering.
+// Similarly, the usage of ''typescript should be ```typescript.
+// Under "Supported Algorithms", consider adding Heap Sort, as it's added to codebase.
+// Good inclusion of a live demo badge.
+// The project structure is well-documented; excellent for new contributors.
+// Consider clarifying the sorting algorithm category includes heap sort to sync with the new code.
+// Looks very developer-friendly and professional.
+// Ensure that anchor links (if any) in TOC correspond with updated headers for navigation support.
+// The new README is now comprehensive and a great guide.
 
-    <title>Algorithm Playground - CallStackSurfer </title>
+    <title>Algorithm Playground</title>
+
+// Minor change: title simplified. Consider if branding "CallStackSurfer" should be kept somewhere for SEO/branding.
+// Otherwise, good for cleaner tab titles.

+import { heapSort } from "../sort/heap-sort/algo";
+import { heapSortVisualizer } from "../sort/heap-sort/visualizer";
+// Correct imports added for heap sort algorithm and visualizer.
+
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSort;
+// Heap sort algorithm function correctly mapped to its SortAlgoType.
+
+      case SortAlgoTypes.HEAP_SORT:
+        return heapSortVisualizer;
+// Visualizer function mapped as well.
+
+// Good to keep the factory updated to enable seamless integration.

+// Implementation of heapSort generator function.
+// Uses generators to yield SortChanges which aligns well with current architecture.
+// Clones input array to avoid mutation, good practice.
+
+// Build max heap phase with heapify calls.
+// Extract elements from the heap phase swaps root with last element, marks sorted, and heapifies remaining.
+// Yields SWAP and SORTED types for state updates.
+// Finally, marks index 0 as sorted after completion.
+// Use of yield* to delegate to heapify enables modular stepwise animation.
+
+// heapify function correctly implements heap property update.
+// Yields PROCESSING type for the node under consideration.
+// Compares left and right children and swaps if needed, then recurses.
+// Good use of TypeScript typings.
+// Clear and easy to follow logic.
+// Suggest adding some comments inside the heapify's swap condition to clarify recursive heapify calls.
+// Overall conforms well to visualization generator expectations.

+// heapSortVisualizer implementation.
+// Takes SortChange and applies updates to UI data items.
+// For SWAP, updates values and borderColor to orange.
+// For PROCESSING, highlights with blue border and background.
+// For SORTED, marks items with green border and background and isSorted flag.
+// Clear, consistent UI state feedback.
+// Use of numeric keys cast with +key is correct.
+// Consider handling potential undefined or out-of-bound errors gracefully, but likely controlled by inputs.
+// Excellent separation of concerns between algorithm logic and visualization.

+    <SortWidget title="Heap Sort" type={SortAlgoTypes.HEAP_SORT} />
+// Correctly adding Heap Sort widget onto the dashboard.
+// Ordering seems appropriate to keep similar sorting algorithms grouped.
+// Consider consistency with whether new algorithms are last or placed in a logical order; here, it's inserted before Bubble Sort.

+  HEAP_SORT = "HEAP_SORT",
+// Adding heap sort enum type.
+// It's consistent with existing naming patterns.
+// Ensure that this enum is used consistently in UI and factory.
```
